xarray/conventions.py

```diff
@@ -514,14 +514,17 @@ def decode_cf(
     decode_times : bool, optional
         Decode cf times (e.g., integers since "hours since 2000-01-01") to
         np.datetime64.
-    decode_coords : bool or {"coordinates", "all"}, optional
+    decode_coords : bool or {"coordinates", "coordinates_strict", "all", "all_strict"}, optional
         Controls which variables are set as coordinate variables:
 
-        - "coordinates" or True: Set variables referred to in the
+        - "coordinates"/"coordinates_strict" or True: Set variables referred to in the
           ``'coordinates'`` attribute of the datasets or individual variables
           as coordinate variables.
-        - "all": Set variables referred to in  ``'grid_mapping'``, ``'bounds'`` and
+        - "all"/"all_strict": Set variables referred to in  ``'grid_mapping'``, ``'bounds'`` and
           other attributes as coordinate variables.
+        When using the "strict"-forms an error is raised, if ``coordinates``-items are
+        missing from the dataset, otherwise a warning is issued and only the
+        available ``coordinates`items are handled.
```

Comment:
```suggestion
```

Reply:
Thanks @dcherian for catching this. I should not commit shortly before leaving.